docs/recipes/patterns/add-dependency-injection.md

```diff
@@ -0,0 +1,56 @@
+> This recipe is not a detailed discussion of the pros and cons of Dependency Injection (DI) compared to other patterns. It simply illustrates how to use it within a SAFE Stack application!
+
+1. Create a class that you wish to inject with a dependency (in this example, we use the built-in `IConfiguration` type that is included in ASP .NET):
+
+    ```fsharp
+    open Microsoft.Extensions.Configuration
+
+    type DatabaseRepository(config:IConfiguration) =
+        member _.SaveDataToDb (text:string) =
+            let connectionString = config["SqlConnectionString"]
+            // Connect to SQL using the above connection string etc.
+            Ok 1
+    ```
+
+    > Instead of functions or modules, DI in .NET and F# only works with classes.
+
+2. Register your type with ASP .NET, typically in the start up / bootstrapper.
+    ```diff
+    ++ open Microsoft.Extensions.DependencyInjection
+
+       application {
+    ++     service_config (fun services -> services.AddSingleton<DatabaseRepository>())
+       }
+    ```
+
+    > [This section](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-8.0#lifetime-and-registration-options) of the official ASP .NET Core article explain the distinction between different lifetime registrations, such as Singleton and Transient.
+
+3. Ensure that your Fable Remoting API can access the `HttpContext` type by using the `fromContext` builder function.
+    ```diff
+    let webApp =
+        Remoting.createApi ()
+    --  |> Remoting.fromValue createFableRemotingApi
+    ++  |> Remoting.fromContext createFableRemotingApi
```

Comment:
This diff misses the |> Remoting.WithRouteBuilder line

Reply:
It just shows the bit that's required, not the whole thing. I'll simply this diff to make it clearer.